kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/lemmas.k

```diff
@@ -37,6 +37,9 @@ module LEMMAS [symbolic]
     rule X in (SetItem(Y) _   ) => true      requires X  ==Int Y [simplification]
     rule X in (SetItem(Y) REST) => X in REST requires X =/=Int Y [simplification]
 
+    rule   ( S:Set |Set SetItem ( X )                      ) |Set SetItem( X ) => S [simplification]
+    rule ( ( S:Set |Set SetItem ( X ) ) |Set SetItem ( Y ) ) |Set SetItem( X ) => S [simplification]
```

Comment:
Are these too ad-hoc now? @ehildenb

Reply:
I'm confused, I don't see why these would be valid. There's nothing here guaranteeing that `X` or `Y` are already in `S`

Reply:
Was it supposed to be `S SetItem(X)` instead of `S |Set SetItem(X)`?

Reply:
Yes, oh wow, so many changes, and I've lost track of what's happening, it's the LHS